test/integration-new/cni/cni_version_test.go

```diff
@@ -0,0 +1,104 @@
+package cni
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/aws/services"
+	"github.com/pkg/errors"
+
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/aws-sdk-go/service/eks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	describeAddonOutput *eks.DescribeAddonOutput
+	err                 error
+	initialCNIVersion   string
+	finalCNIVersion     string
+)
+
+var _ = Describe("test cluster upgrade/downgrade", func() {
+
+	It("should apply initial addon version successfully", func() {
+		By("getting initial cni version")
+		if len(f.Options.InitialCNIVersion) == 0 {
+			err = errors.Errorf("%s must be set!", "initial-version")
+		}
+		Expect(err).ToNot(HaveOccurred())
+		initialCNIVersion = f.Options.InitialCNIVersion
+		ApplyAddOn(initialCNIVersion)
+
+	})
+
+	Context("when testing host networking on initial version", func() {
+		HostNetworkingTest()
+		PodTrafficTest()
+		ServiceConnectivityTest()
+	})
+
+	It("should apply final addon version successfully", func() {
+		By("getting final cni version")
+		if len(f.Options.FinalCNIVersion) == 0 {
+			err = errors.Errorf("%s must be set!", "final-version")
+		}
+		Expect(err).ToNot(HaveOccurred())
+		finalCNIVersion = f.Options.FinalCNIVersion
+		ApplyAddOn(finalCNIVersion)
+
+	})
+
+	Context("when testing host networking on final version", func() {
+		HostNetworkingTest()
+		PodTrafficTest()
+		ServiceConnectivityTest()
```

Comment:
I think these 3 functions are 3 different kinds of tests, so should be separated out in 3 Contexts ? would like to know your thoughts